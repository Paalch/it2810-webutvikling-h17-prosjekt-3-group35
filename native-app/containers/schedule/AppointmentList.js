import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import Appointment from "./Appointment";
import moment from 'moment';


export default class AppointmentList extends React.Component {
    constructor(props) {
        super(props);
        this.deleteAppointment = this.deleteAppointment.bind(this);
    }

    deleteAppointment(id) {
        const appointments = this.props.appointments.slice().filter((a) => a.id !== id);
        this.props.onDeleteAppointment(appointments);
    }

    componentDidMount() {
        this.interval = setInterval(() => {
            const appointments = this.props.appointments.slice().filter((a) => a.date.isAfter(moment().subtract(1, "hours")));
            this.props.onDeleteAppointment(appointments);

        }, 60000);
    }

    componentWillUnmount() {
        clearInterval(this.interval);
    }

    render() {
        const appointments = this.props.appointments.sort((a, b) => a.date.isAfter(b.date)).map((appointment, i) =>
            <Appointment
                key={i}
                appointment={appointment}
                text={appointment.details}
                id={appointment.id}
                time={appointment.date.format("DD.MM.YYYY") + " " + appointment.time.format("HH:mm")}
                color={appointment.color}
                appointments={appointments}
                deleteAppointment={this.deleteAppointment}
                receiveAppointment={this.receiveAppointment}/>);
        return (
            <View>
                <View>
                    {appointments}
                </View>
            </View>
        );
    }
}

