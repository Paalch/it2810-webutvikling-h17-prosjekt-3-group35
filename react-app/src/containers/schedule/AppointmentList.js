import React, {Component} from 'react'
import Appointment from "./Appointment";



class AppointmentList extends Component {

  constructor(props) {
    super(props);
    this.state = {
      appointments: this.props.appointments
    };
    this.receiveAppointment = this.receiveAppointment.bind(this);
  }

  receiveAppointment(appointments) {
    this.state = {
      appointments: appointments
    };
    this.props.onAppointments(this.state.appointments);
  }

  render() {
    const appointments = (this.state.appointments).sort((a, b) => a[2] > b[2]).map((apps, i) =>
      <Appointment
          content={apps}
        text={apps[1]} moment={apps[2]} key={i} id={apps[3]} time={apps[0]} color={apps[4]} appointments={this.state.appointments}
        receiveAppointment={this.receiveAppointment}/>);
    return (
      <div>
        <div>
          {appointments}
        </div>
      </div>
    );

  }
}

export default AppointmentList;