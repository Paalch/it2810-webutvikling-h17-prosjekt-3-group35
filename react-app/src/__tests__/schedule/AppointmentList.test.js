import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import renderer from 'react-test-renderer'
import AppointmentList from "../../containers/schedule/AppointmentList";
import moment from 'moment'

Enzyme.configure({ adapter: new Adapter() })


describe('Checks that the appointmentList component renders correctly', () => {
    let appointment = {
        date: moment(),
        time: moment(),
        details: 'sometext',
        id: 1,
        style: 'primary'
    };

    it('Checks that appointmentList renders', () => {
        const rendered = renderer.create(
            <AppointmentList
                appointments={[appointment]}
                onDeleteAppointment={ () => {return false}}/>
        );
        expect(rendered.toJSON()).toMatchSnapshot()
    });

});
