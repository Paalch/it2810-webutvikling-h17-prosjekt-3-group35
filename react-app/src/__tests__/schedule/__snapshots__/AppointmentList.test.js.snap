// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks that the appointmentList component renders correctly Checks that appointmentList renders 1`] = `
<div>
  <div>
    <div
      className="appointment-container"
    >
      <div
        className="panel panel-primary"
        id={undefined}
      >
        <div
          className="panel-heading"
        >
          <div
            className="panel-title"
          >
            <div
              className="row"
            >
              <div
                className="col-xs-8"
              >
                <p>
                  18.10.2017 21:23
                </p>
              </div>
              <div
                className="col-xs-4"
              >
                <button
                  className="delete-appointment btn btn-link"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  X
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="panel-body"
        >
          sometext
        </div>
      </div>
    </div>
  </div>
</div>
`;
